### ISSUE TITLE
Enumerate continuous RVs using Monte Carlo or quadrature

### ISSUE THREAD

**Issue by fritzo** [uber/pyro#811]
After #776 merges, it should be easy to extend parallel enumeration from discrete distributions to continuous distributions using either Monte Carlo sampling or quadrature. Using the `infer` dict and `InferConfigPoutine`, it should be easy to add options to sample sites like
```py
infer={"enumerate": {"type": "monte_carlo", "num_samples": 10}}
```
and
```py
 infer={"enumerate": {"type": "quadrature", "grid": torch.arange(-5,6,1)}}
```
This logic could then be handled in `EnumeratePoutine._pyro_sample()` analogously to its handling of discrete sites, though this behavior might additionally need to set `site["scale"]` if weighted quadrature is used.

**Comment by ngoodman** [comment]
this is a fun idea. we did something similar for continuous rvs in webpl, but not in parallel.

(it's also related to the tree/AIS this that @mhw32 is about to start on.)

**Comment by fritzo** [comment]
@eb8680 as per discussion today, Monte Carlo is already possible, and quadrature and antithetical sampling should be achievable by combining `infer={"num_samples": 100}` #963 with a special `ReshapedDistribution` e.g. `QuadratureReshapedDistribution` or `AntitheticalReshapedDistribution`, i.e. we should need no extra machinery in `TraceEnum_ELBO`.

**Comment by fritzo** [comment]
Closing in favor of a more targeted issue, since #963 solved most of this issue.
